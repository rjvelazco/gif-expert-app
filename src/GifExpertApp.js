import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {

    // const [categories, setCategories] = useState(['Fire Force']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpert</h2>
            <AddCategory setCategories={setCategories} />
            <hr />
            <div>
                {categories.map((category) => (
                    <GifGrid
                        key={ category }
                        category={category}
                    />
                ))}
            </div>

        </>
    )
}

export default GifExpertApp;

GifExpertApp.propTypes = {
    GifExpertApp: PropTypes.array
}