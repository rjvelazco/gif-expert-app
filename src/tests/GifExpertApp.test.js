import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

// Component
import GifExpertApp from '../GifExpertApp';

describe('test on <GifExpertApp /> component', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('It should render <GifExpertApp /> component correctly', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('It should show a list of categories', () => {
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })

});