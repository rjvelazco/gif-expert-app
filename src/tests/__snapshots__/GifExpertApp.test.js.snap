// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifExpertApp /> component It should render <GifExpertApp /> component correctly 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div />
</Fragment>
`;

exports[`test on <GifExpertApp /> component It should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpert
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;
